src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -93,6 +100,76 @@ class GolangSubsystem(Subsystem):
         advanced=True,
     )
 
+    cgo_enabled = BoolOption(
+        "--cgo-enabled",
+        default=False,
+        help=softwrap(
+            """\
+            Enable Cgo, which allows calling C code from Go.
+
+            See https://go.dev/blog/cgo and https://pkg.go.dev/cmd/cgo for additional information.
+            """
+        ),
+    )
+
+    _cgo_tool_search_paths = StrListOption(
+        default=["<PATH>"],
+        help=softwrap(
+            """
+            A list of paths to search for tools needed by CGo (e.g., gcc, g++).
+
+            Specify absolute paths to directories with tools needed by CGo , e.g. `/usr/bin`.
+            Earlier entries will be searched first.
+
+            The following special strings are supported:
+
+              * `<PATH>`, the contents of the PATH environment variable
+            """
+        ),
+    )
+
+    cgo_gcc_binary_name = StrOption(
+        default="gcc",
+        advanced=True,
+        help="Name of the tool to use to compile C code included via CGo in a Go package.",
+    )
+
+    cgo_gxx_binary_name = StrOption(
+        default="g++",
+        advanced=True,
+        help="Name of the tool to use to compile C++ code included via CGo in a Go package.",
+    )
+
+    cgo_fortran_binary_name = StrOption(
+        default="gfortran",
+        advanced=True,
+        help="Name of the tool to use to compile fortran code included via CGo in a Go package.",
+    )
+
+    cgo_default_cflags = StrListOption(
```

Comment:
Bikeshed: In other Subsystems, we don't tend to use the term "default" in the option name. Is this intended to indicate something specific to `cgo`?

Reply:
With `go`, it does not use the `-g -O2` default for cflags if `CGO_CFLAGS` is set. But for Pants, we won't be looking at `CGO_CFLAGS` so I agree makes more sense to just call this `cgo_cflags` (and similar change to the other options).

Reply:
👍🏽